Add B2Share support #88

micafer opened the pull request.

J535D165 commented:
It would be very cool to add support for this!

Can you rebase?

J535D165 commented:
It might also be nice to split the filter-files extension from the PR so that we can have a closer look at the design. It looks promising.

micafer commented:
Code cleaned.

J535D165 commented on datahugger/base.py:
```diff
@@ -284,6 +284,7 @@ def _get_files_recursive(self, url, folder_name=None, base_url=None):
 
         # get the data from URL
         res = requests.get(url)
+        res.raise_for_status()
```
```suggestion
```

Introduced in #80

J535D165 commented on datahugger/services.py:
```diff
@@ -320,7 +320,7 @@ class MendeleyDataset(DatasetDownloader):
     """Downloader for OSF repository."""
 
-    REGEXP_ID = r"osf\.io\/(?P<record_id>.*)/"
+    REGEXP_ID = r"osf\.io\/(?P<record_id>[^\/]*)\/{0,1}"
```
Interesting catch. I'm fine with keeping this in this PR.

J535D165 commented:
Apparently, the test fails because I removed the raise_for_status d8a4a4915fead78ef16c8d7842bc0fba638449bc. I'm a bit confused.

J535D165 commented on datahugger/services.py:
```diff
@@ -425,3 +425,25 @@ class SeaNoeDataset(DatasetDownloader):
+    API_URL_META = "{api_url}records/{record_id}"
+    META_FILES_JSONPATH = "files[*]"
+
+    # paths to file attributes
+    ATTR_NAME_JSONPATH = "key"
+    ATTR_SIZE_JSONPATH = "size"
+    ATTR_HASH_JSONPATH = "checksum"
+    ATTR_HASH_TYPE_VALUE = "md5"
+
+    def _get_attr_link(self, record, base_url=None):
+        return f"{base_url}/files/{self._params['record_id']}/{record['key']}"
```
```suggestion

    # paths to file attributes
    ATTR_NAME_JSONPATH = "key"
    ATTR_FILE_LINK_JSONPATH = "ePIC_PID"
    ATTR_SIZE_JSONPATH = "size"
    ATTR_HASH_JSONPATH = "checksum"
    ATTR_HASH_TYPE_VALUE = "md5"
```

This seems to work. I'm clueless about whats going on.

J535D165 commented:
Now it passes. Did they change something to their API since the commit of @micafer 5 hours ago...? 

Nevertheless, thanks a lot, @micafer, for another cool contribution!

micafer commented:
> Now it passes. Did they change something to their API since the commit of @micafer 5 hours ago...?

It should not have changed.

micafer commented:
> Apparently, the test fails because I removed the raise_for_status [d8a4a49](https://github.com/J535D165/datahugger/commit/d8a4a4915fead78ef16c8d7842bc0fba638449bc). I'm a bit confused.

It should not affect the tests.
I added to improve the error message shown to the user in case of HTTP error. Without it the HTTP error is hidden and the error message shown is a bit confusing,

J535D165 commented:
It works now! I tested it on the benchmark dataset. Everything looks good.
